packages/web3.js-plugin-fee-estimation/src/plugin.ts

```diff
@@ -0,0 +1,337 @@
+import Web3, {
+  BlockNumberOrTag,
+  BlockTags,
+  Contract,
+  DataFormat,
+  DEFAULT_RETURN_FORMAT,
+  FMT_BYTES,
+  FMT_NUMBER,
+  Numbers,
+  Transaction,
+  Web3PluginBase,
+} from 'web3'
+import { TransactionFactory, TxData } from 'web3-eth-accounts'
+import { estimateGas, formatTransaction } from 'web3-eth'
+import {
+  gasPriceOracleABI,
+  gasPriceOracleAddress,
+} from '@eth-optimism/contracts-ts'
+import { RLP } from '@ethereumjs/rlp'
+
+export default class OptimismFeeEstimationPlugin extends Web3PluginBase {
+  public pluginNamespace = 'op'
+
+  private _gasPriceOracleContract:
+    | Contract<typeof gasPriceOracleABI>
+    | undefined
+
+  /**
+   * Retrieves the current L2 base fee
+   * @param {DataFormat} [returnFormat=DEFAULT_RETURN_FORMAT] - The web3.js format object that specifies how to format number and bytes values
+   * @returns {Promise<bigint>} - The L2 base fee as a BigInt by default, but {returnFormat} determines type
+   * @example
+   * const baseFeeValue: bigint = await web3.op.getBaseFee();
+   * @example
+   * const numberFormat = { number: FMT_NUMBER.NUMBER, bytes: FMT_BYTES.HEX }
+   * const baseFeeValue: number = await web3.op.getBaseFee(numberFormat);
+   */
+  public async getBaseFee<
+    ReturnFormat extends DataFormat = typeof DEFAULT_RETURN_FORMAT
+  >(returnFormat?: ReturnFormat) {
```

Comment:
nit: for libraries prefer being explicit about the return type. This helps guarantee we don't accidentally make a breaking change.

Reply:
```suggestion
  >(returnFormat?: ReturnFormat): Promise<bigint> {
```

Reply:
Ah this is actually pretty complicated though since the return type depends on the return format. What do you think about being more modular and only returning the result of the call and expecting the user to call Web3.utils.format themselves? Simplifies the API, simplifies the type infererence, more modular, more of a 1 to 1 match with the underlying api.

Alternatively force BigInt

Reply:
Actually nvm I see that estimateGas from web3.js does teh same pattern. We should match what web3.js does 1to1 so ignore the above suggestion. And the return type is complicated enough where we are likely best off just inferring it rather than being explicit